Set up backups for umbriel

jfly opened the pull request:
There are multiple commits here, I suggest reading them separately.

jfly committed:
Set up backups for umbriel

This does not actually back up any files/folders yet, it just sets up
the infrastructure for doing so. This is necessary because umbriel is
now stateful (now that we use it for freescout).

jfly commented on non-critical-infra/hosts/umbriel/default.nix:
```diff
@@ -44,5 +50,39 @@
+  };
+
+  sops.secrets.backup-secret = {
+    sopsFile = ../../secrets/backup-secret.umbriel; # <<< TODO: how to generate >>>
```
@mweinelt, I read through https://github.com/NixOS/infra/pull/345, but wasn't able to figure out how you generated this for caliban. Can you point me in the right direction?

mweinelt replied:
Probably https://github.com/NixOS/infra/pull/345#issuecomment-1914977208 with init instead of list.

jfly replied:
Ooh, turns out that wasn't necessary because we have [`doInit = true`](https://github.com/jfly/infra/blob/c21bc89ef5fefa200913c8581bef94217ebe78dd/modules/backup.nix#L158-L159)

jfly committed:
Backup simple nixos mailserver + freescout

Now that we support storing email for IMAP accounts and freescout for
collaborative inboxes, umbriel has state that we care about and should
back up.

I made a best guess here about what's important to back up for
freescout. See this upstream issue where I asked for confirmation that I
got it right: <https://cyberchaos.dev/e1mo/freescout-nix-flake/-/issues/7>

For SNM, I followed their [Backup
Guide](https://nixos-mailserver.readthedocs.io/en/latest/backup-guide.html).
I opted to only back up our `mailDirectory`. We haven't enabled
`enableManageSieve`. I'm not sure if we should back up our spam and ham
training data. We don't need to back up our dkim data, as those secrets
are encrypted and version controlled.

refs: https://github.com/NixOS/infra/issues/700

jfly commented:
I went ahead and deployed this, it works!

```
[root@umbriel:~]# systemctl status borgbackup-job-state.service
○ borgbackup-job-state.service - BorgBackup job state
     Loaded: loaded (/etc/systemd/system/borgbackup-job-state.service; linked; preset: ignored)
     Active: inactive (dead) since Mon 2025-09-22 21:51:19 UTC; 4min 57s ago
   Duration: 17.570s
 Invocation: edf437ac2e984ed68b1e0b28ec2bb184
TriggeredBy: ● borgbackup-job-state.timer
    Process: 2471867 ExecStart=/nix/store/apfcwhhfqi5b171bp5w51iwic44dm2if-unit-script-borgbackup-job-state-start/bin/borgbackup-job-state-start (code=exited, status=0/SUCCESS)
   Main PID: 2471867 (code=exited, status=0/SUCCESS)
         IP: 1.2M in, 376.1M out
         IO: 237.3M read, 0B written
   Mem peak: 80.6M
        CPU: 11.867s

Sep 22 21:51:16 umbriel borgbackup-job-state-start[2471895]: Utilization of max. archive size: 0%
Sep 22 21:51:16 umbriel borgbackup-job-state-start[2471895]: ------------------------------------------------------------------------------
Sep 22 21:51:16 umbriel borgbackup-job-state-start[2471895]:                        Original size      Compressed size    Deduplicated size
Sep 22 21:51:16 umbriel borgbackup-job-state-start[2471895]: This archive:              607.13 MB            487.22 MB            392.50 MB
Sep 22 21:51:16 umbriel borgbackup-job-state-start[2471895]: All archives:              607.12 MB            487.22 MB            392.72 MB
Sep 22 21:51:16 umbriel borgbackup-job-state-start[2471895]:                        Unique chunks         Total chunks
Sep 22 21:51:16 umbriel borgbackup-job-state-start[2471895]: Chunk index:                    2414                 2827
Sep 22 21:51:16 umbriel borgbackup-job-state-start[2471895]: ------------------------------------------------------------------------------
Sep 22 21:51:19 umbriel systemd[1]: borgbackup-job-state.service: Deactivated successfully.
Sep 22 21:51:19 umbriel systemd[1]: borgbackup-job-state.service: Consumed 11.867s CPU time, 80.6M memory peak, 237.3M read from disk, 1.2M incoming IP traffic, 376.1M outgoing IP traffic.
```

mweinelt commented on non-critical-infra/modules/mailserver/default.nix:
```diff
@@ -28,6 +28,9 @@
   };
 
+  # https://nixos-mailserver.readthedocs.io/en/latest/backup-guide.html
+  services.backup.includes = [ config.mailserver.mailDirectory ];
```
That leaves out
- /var/sieve
- /var/dkim
- /var/lib/redis-rspamd

Is that a concious design choice/

mweinelt replied:
That covers dkim and sieve. I'm not sure we even train the bayes filter, so we might as well leave it at that.
